lib/view_component/compiler.rb

```diff
@@ -146,18 +146,16 @@ def matching_views_in_source_location
       source_location = component_class.source_location
       return [] unless source_location
 
-      location_without_extension = source_location.chomp(File.extname(source_location))
```

Comment:
This is technically a breaking change and now ignores the component's file name in template lookups. This is so we can define components inside of components.

There is work to be done on how to find the nested component's template though.

e.g.

```
class ModalComponent < ViewComponent
  class FooterComponent < ViewComponent
  end
end
```

Where do we look for the `FooterComponent` template files? My first thought is a subdirectory, so `modal_component/footer_component.html.erb`

Reply:
Yes, let's go with the subdirectory 👍🏻